python-wrapper/src/neo4j_viz/gql_create.py

```diff
@@ -0,0 +1,337 @@
+            props["__labels"] = props["labels"]
+        props["labels"] = sorted(label_list)
+
+        return final_alias, top_level, props
+
+    nodes = []
+    relationships = []
+    alias_to_id = {}
+    anonymous_count = 0
+
+    query = re.sub(r"(?i)^create\s*", "", query, count=1).rstrip(";").strip()
+    parts = []
+    paren_level = 0
+    bracket_level = 0
+    current: list[str] = []
+    for i, char in enumerate(query):
+        if char == "(":
+            paren_level += 1
+        elif char == ")":
+            paren_level -= 1
+            if paren_level < 0:
+                snippet = _get_snippet(query, i)
+                raise ValueError(f"Unbalanced parentheses near: `{snippet}`.")
+        if char == "[":
+            bracket_level += 1
+        elif char == "]":
+            bracket_level -= 1
+            if bracket_level < 0:
+                snippet = _get_snippet(query, i)
+                raise ValueError(f"Unbalanced square brackets near: `{snippet}`.")
+        if char == "," and paren_level == 0 and bracket_level == 0:
+            parts.append("".join(current).strip())
+            current = []
+        else:
+            current.append(char)
+    parts.append("".join(current).strip())
+    if paren_level != 0:
+        snippet = _get_snippet(query, len(query) - 1)
+        raise ValueError(f"Unbalanced parentheses near: `{snippet}`.")
+    if bracket_level != 0:
+        snippet = _get_snippet(query, len(query) - 1)
+        raise ValueError(f"Unbalanced square brackets near: `{snippet}`.")
+
+    node_pattern = re.compile(r"^\(([^)]+)\)$")
+    rel_pattern = re.compile(r"^\(([^)]+)\)-\s*\[\s*:(\w+)\s*(\{[^}]*\})?\s*\]->\(([^)]+)\)$")
+
+    node_top_level_keys = set(Node.model_fields.keys())
+    node_top_level_keys.remove("id")
+
+    rel_top_level_keys = set(Relationship.model_fields.keys())
+    rel_top_level_keys.remove("id")
+    rel_top_level_keys.remove("source")
+    rel_top_level_keys.remove("target")
+
+    empty_set: set[str] = set()
+
+    for part in parts:
+        node_m = node_pattern.match(part)
+        if node_m:
+            alias_labels_props = node_m.group(1).strip()
+            alias, top_level, props = parse_labels_and_props(alias_labels_props, node_top_level_keys)
+            if not alias:
+                alias = f"_anon_{anonymous_count}"
+                anonymous_count += 1
+            if alias not in alias_to_id:
+                alias_to_id[alias] = str(uuid.uuid4())
+            nodes.append(Node(id=alias_to_id[alias], **top_level, properties=props))
+        else:
+            rel_m = rel_pattern.match(part)
+            if rel_m:
+                left_node = rel_m.group(1).strip()
+                rel_type = rel_m.group(2).replace(":", "").strip()
+                right_node = rel_m.group(4).strip()
+
+                left_alias, _, _ = parse_labels_and_props(left_node, empty_set)
+                if not left_alias or left_alias not in alias_to_id:
```

Comment:
so alias is mandatory?

Reply:
For relationships, yes? Otherwise you cannot determine which the start and end nodes are

Reply:
they could be anonymous nodes. Lets say you only have `()-->()` this could create two nodes and 1 relationship.
Its something simple we used in tests in the past.
Also cypher allows you the create new nodes using `CREATE (), ()`

Reply:
done